drivers/sensor/max17055/max17055.c

```diff
@@ -222,31 +353,57 @@ static int max17055_gauge_init(const struct device *dev)
 		return -EINVAL;
 	}
 
-	return 0;
+	if (max17055_reg_read(priv, STATUS, &tmp)) {
+		return -EIO;
+	}
+
+	if (!(tmp & STATUS_POR)) {
+		LOG_DBG("No POR event detected - skip device configuration");
+		return 0;
+	}
+
+	/* Wait for FSTAT_DNR to be cleared */
+	tmp = FSTAT_DNR;
+	while (tmp & FSTAT_DNR)
+		max17055_reg_read(priv, FSTAT, &tmp);
+
+	if (max17055_init_config(priv, config)) {
+		return -EIO;
+	}
+
+	/* Clear PowerOnReset bit */
+	if (max17055_reg_read(priv, STATUS, &tmp)) {
+		return -EIO;
+	}
+
+	tmp &= ~STATUS_POR;
+	return max17055_reg_write(priv, STATUS, tmp);
 }
 
 static const struct sensor_driver_api max17055_battery_driver_api = {
 	.sample_fetch = max17055_sample_fetch,
 	.channel_get = max17055_channel_get,
 };
 
-#define MAX17055_INIT(index)						\
-	static struct max17055_data max17055_driver_##index;		\
-									\
-	static const struct max17055_config max17055_config_##index = { \
-		.bus_name = DT_INST_BUS_LABEL(index),			\
-		.rsense_mohms = DT_INST_PROP(index, rsense_mohms),	\
-		.design_voltage = DT_INST_PROP(index, design_voltage),	\
-		.desired_voltage = DT_INST_PROP(index, desired_voltage), \
-		.desired_charging_current =				\
-			DT_INST_PROP(index, desired_charging_current),	\
-	};								\
-									\
-	DEVICE_DT_INST_DEFINE(index, &max17055_gauge_init,		\
-			    device_pm_control_nop,			\
-			    &max17055_driver_##index,			\
-			    &max17055_config_##index, POST_KERNEL,	\
-			    CONFIG_SENSOR_INIT_PRIORITY,		\
-			    &max17055_battery_driver_api)
+#define MAX17055_INIT(index)								   \
+	static struct max17055_data max17055_driver_##index;				   \
+											   \
+	static const struct max17055_config max17055_config_##index = {			   \
+		.bus_name = DT_INST_BUS_LABEL(index),					   \
+		.design_capacity = DT_INST_PROP_OR(index, design_capacity, 1500),	   \
+		.design_voltage = DT_INST_PROP(index, design_voltage),			   \
+		.desired_charging_current = DT_INST_PROP(index, desired_charging_current), \
+		.desired_voltage = DT_INST_PROP(index, desired_voltage),		   \
+		.i_chg_term = DT_INST_PROP_OR(index, i_chg_term, 100),			   \
+		.rsense_mohms = DT_INST_PROP(index, rsense_mohms),			   \
+		.v_empty = DT_INST_PROP_OR(index, v_empty, 3300),			   \
```

Comment:
JUst my curiousity. Is using DT_INST_PROP_OR() same as declaring '3300' in the dts binding as default? If so, why choosing one vs the other? Just for my better knowledge. And some brainstorming as well...

Reply:
I've no idea - I wasn't aware that declaring defaults in the dts binding was possible.
Happy to give the other option a go.

Reply:
@galak @mbolivar-nordic do you have a preferred way to do this?

Reply:
> @galak @mbolivar-nordic do you have a preferred way to do this?

A binding with `default` is justified if it's the PoR value for the sensor.  The full rules are [here](https://github.com/zephyrproject-rtos/zephyr/blob/58a0ba6dbb28413bbad19d2eb41e1f3524b0e4fa/dts/binding-template.yaml#L217) which is included in the bindings doc page.

Reply:
Thank you @mbolivar-nordic - this case is actually explicitly mentioned in the documentation you've linked:

https://github.com/zephyrproject-rtos/zephyr/blob/58a0ba6dbb28413bbad19d2eb41e1f3524b0e4fa/dts/binding-template.yaml#L226-L231

I'd originally left this as not required as it would be a breaking change, but I'll switch these fields to required as per the above.